core/src/main/scala/caliban/execution/Field.scala

```diff
@@ -35,7 +36,33 @@ object Field {
     rootType: RootType
   ): Field = {
     def loop(selectionSet: List[Selection], fieldType: __Type): Field = {
-      val fieldList = List.newBuilder[Field]
+      val fieldList  = ArrayBuffer.empty[Field]
```

Comment:
Should we give this a size hint as well? I'd expect that to provide a minor improvement in 99% of cases where people don't duplicate their fields (and really only penalizes us in the pathological case).

Reply:
But we don't know the size without going through fragments recursively.